streams/src/main/java/org/apache/kafka/streams/KeyValue.java

```diff
@@ -42,4 +44,27 @@ public KeyValue(K key, V value) {
     public String toString() {
         return "KeyValue(" + key + ", " + value + ")";
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == null)
```

Comment:
Nitpick:  We don't need to explicitly check for `other == null` -- `instanceof` (which is required for the casting logic anyways) will return false if its first argument is null.